fem/bilinearform.hpp

```diff
@@ -359,20 +422,67 @@ class BilinearForm : public Matrix
 
    virtual void Update(FiniteElementSpace *nfes = NULL);
 
-   /// (DEPRECATED) Return the FE space associated with the BilinearForm.
+   /// (DEPRECATED) Return the FE space associated with the FABilinearForm.
    /** @deprecated Use FESpace() instead. */
    FiniteElementSpace *GetFES() { return fes; }
 
-   /// Return the FE space associated with the BilinearForm.
+   /// Return the FE space associated with the FABilinearForm.
    FiniteElementSpace *FESpace() { return fes; }
    /// Read-only access to the associated FiniteElementSpace.
    const FiniteElementSpace *FESpace() const { return fes; }
 
    /// Sets diagonal policy used upon construction of the linear system
-   void SetDiagonalPolicy(DiagonalPolicy policy);
+   void SetDiagonalPolicy(Matrix::DiagonalPolicy policy);
 
    /// Destroys bilinear form.
-   virtual ~BilinearForm();
+   virtual ~FABilinearForm();
+};
+
+// *****************************************************************************
+// *
+// *****************************************************************************
+class BilinearForm
```

Comment:
This new definition of `BilinearForm` will break pretty much any existing code based on MFEM. I think we need to be backward compatible.

Reply:
If we want backward compatibility, then I agree that we need to keep the original names and have another way to switch the FA|PA|MF options.

Reply:
`FiniteElementOperator`? I always was a bit confused by the `BilinearForm` name anyway, since it's a lot more than a bilinear form, but also not really a bilinear form since you don't apply it to 2 Vectors...